AbletonMCP_Remote_Script/__init__.py

```
@@ -548,6 +609,120 @@ def _set_clip_name(self, track_index, clip_index, name):
             self.log_message("Error setting clip name: " + str(e))
             raise
     
+    def _duplicate_clip(self, source_track_index, source_clip_index, dest_track_index, dest_clip_index):
+        """Duplicate a clip from one slot to another"""
+        try:
+            # Validate source track and clip
+            if source_track_index < 0 or source_track_index >= len(self._song.tracks):
+                raise IndexError("Source track index out of range")
```

Ruff (0.14.6)

Abstract `raise` to an inner function

(TRY301)